skate-plugin/project-gen/src/jvmTest/kotlin/slack/tooling/projectgen/CircuitComponentTest.kt

```diff
@@ -0,0 +1,207 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.tooling.projectgen
+
+import com.google.common.truth.Truth.assertThat
+import java.io.StringWriter
+import org.junit.Test
+import slack.tooling.projectgen.circuitgen.AssistedInjectionConfig
+import slack.tooling.projectgen.circuitgen.CircuitGenClassNames.Companion.APP_SCOPE
+import slack.tooling.projectgen.circuitgen.CircuitGenClassNames.Companion.USER_SCOPE
+import slack.tooling.projectgen.circuitgen.CircuitPresenter
+import slack.tooling.projectgen.circuitgen.CircuitScreen
+import slack.tooling.projectgen.circuitgen.CircuitTest
+import slack.tooling.projectgen.circuitgen.CircuitViewModel
+
+class CircuitComponentTest {
+  @Test
+  fun testGenerateCircuitScreen() {
+    val component = CircuitScreen()
+    val resultSpec = component.generate("com.example.feature", "FooScreen")
+    val stringWriter = StringWriter()
+    resultSpec.writeTo(stringWriter)
+    val expectedContent =
+      """
+      package com.example.feature
+
+      import androidx.compose.runtime.Immutable
+      import com.slack.circuit.runtime.CircuitUiEvent
+      import com.slack.circuit.runtime.CircuitUiState
+      import com.slack.circuit.runtime.screen.Screen
+      import kotlin.Unit
+      import kotlinx.parcelize.Parcelize
+
+      @Parcelize
+      public class FooScreen : Screen {
+        public data class State(
+          public val eventSink: Event.() -> Unit = {},
+        ) : CircuitUiState
+
+        @Immutable
+        public sealed interface Event : CircuitUiEvent
+      }
+
+      """
+        .trimIndent()
+    assertThat(stringWriter.toString()).isEqualTo(expectedContent)
+  }
+
+  @Test
+  fun testGenerateCircuitPresenterNoAssistedInjection() {
+    val component = CircuitPresenter(AssistedInjectionConfig(), setOf(USER_SCOPE))
+    val resultSpec = component.generate("com.example.feature", "FooPresenter")
+    val stringWriter = StringWriter()
+    resultSpec.writeTo(stringWriter)
+    val expectedContent =
+      """
+      package com.example.feature
+
+      import androidx.compose.runtime.Composable
+      import com.slack.circuit.codegen.annotations.CircuitInject
+      import com.slack.circuit.runtime.Navigator
+      import com.slack.circuit.runtime.presenter.Presenter
+      import slack.di.UserScope
+
+      public class FooPresenter(
+        private val screen: FooScreen,
+        private val navigator: Navigator,
+      ) : Presenter<FooScreen.State> {
+        @Composable
+        override fun present(): FooScreen.State {
+          TODO("Implement me!")
+        }
+
+        @CircuitInject(
+          FooScreen::class,
+          UserScope::class,
+        )
+        public fun interface Factory {
+          public fun create(screen: FooScreen, navigator: Navigator): FooPresenter
+        }
```

Comment:
If there's no assisted parameters, we should not generate a factory nor add the screen/navigator params. We should instead
- Put the CircuitInject annotation on the presenter class itself
- Create a primary constructor annotated with `@Inject`, i.e. `@Inject constructor()`.

Reply:
added the case when there's no assisted params
https://github.com/slackhq/slack-gradle-plugin/pull/799/commits/0e7541ab15a1b946d3175ad56082a28ccbbf8eda